packages/beeq/src/components/page-title/bq-page-title.tsx

```diff
@@ -0,0 +1,170 @@
+
+import { hasSlotContent } from '../../shared/utils';
+
+/**
+ * @part wrapper - The wrapper container `<div>` of the element inside the shadow DOM.
+ * @part back - The container `<div>` that wraps the page title back icon button.
+ * @part btn-back - The back navigation button.
+ * @part title - The container `<div>` that wraps the page title content.
+ * @part icon - The `<bq-icon>` element used to render a predefined back navigation icon for page title.
+ * @part prefix - The `<div>` page title element that acts as prefix slot container.
+ * @part suffix - The `<div>` page title element that acts as suffix slot container.
+ * @part sub-title - The `<div>` page title element that acts as sub-title slot container.
+ */
+@Component({
+  tag: 'bq-page-title',
+  styleUrl: './scss/bq-page-title.scss',
+  shadow: true,
+})
+export class BqPageTitle {
+  // Own Properties
+  // ====================
+
+  private prefixElem: HTMLElement;
+  private suffixElem: HTMLElement;
+  private subTitleElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqPageTitleElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private hasPrefix = false;
+  @State() private hasSuffix = false;
+  @State() private hasSubTitle = false;
+
+  // Public Property API
+  // ========================
+
+  // Prop lifecycle events
+  // =======================
+
+  /** If true, the page title back button will be shown */
+  @Prop({ reflect: true }) haveBackNavigation: boolean;
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when page title navigation button loses focus */
+  @Event() bqBlur: EventEmitter<HTMLBqPageTitleElement>;
+
+  /** Handler to be called when page title navigation button is clicked */
+  @Event() bqClick: EventEmitter<HTMLBqPageTitleElement>;
+
+  /** Handler to be called when page title navigation button is focused */
+  @Event() bqFocus: EventEmitter<HTMLBqPageTitleElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleClick = () => {
+    this.bqClick.emit(this.el);
+  };
+
+  private handleBlur = () => {
+    this.bqBlur.emit(this.el);
+  };
+
+  private handleFocus = () => {
+    this.bqFocus.emit(this.el);
+  };
```

Comment:
We should stop propagation here, and probably instead of emitting flat events, to be more meaningful:
* `bqBackClick`
* `bqBackBlur`
* `bqBackFocus